plugins/main/bridge/bridge.go

```diff
@@ -386,14 +386,23 @@ func cmdDel(args *skel.CmdArgs) error {
 		return nil
 	}
 
-	var ipn *net.IPNet
-	err = ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
-		var err error
-		ipn, err = ip.DelLinkByNameAddr(args.IfName, netlink.FAMILY_V4)
+	// Check whether the device exists before trying to delete it
+	devErr := ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
+		_, err := netlink.LinkByName(args.IfName)
 		return err
 	})
-	if err != nil {
-		return err
+
+	// Only try to delete the device if it could be found in the first place.
+	var ipn *net.IPNet
+	if devErr == nil {
+		err = ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
+			var err error
+			ipn, err = ip.DelLinkByNameAddr(args.IfName, netlink.FAMILY_V4)
+			return err
+		})
+		if err != nil {
+			return err
+		}
 	}
 
 	if n.IPMasq {
```

Comment:
If the iptables operations below also error - I would assume in some cases we would want to also ignore (e.g. chain does not exist?)